src/extension/background-script/actions/lnurl/auth.ts

```diff
@@ -0,0 +1,178 @@
+    .where("host")
+    .equalsIgnoreCase(message.origin.host)
+    .first();
+
+  // we have the check the unlock status manually. The account can still be locked
+  // If it is locked we must show a prompt to unlock
+  const isUnlocked = state.getState().password !== null;
+
+  let loginStatus;
+  // check if there is a publisher and lnurlAuth is enabled,
+  // otherwise we we prompt the user
+  if (isUnlocked && allowance && allowance.enabled && allowance.lnurlAuth) {
+    loginStatus = { confirmed: true, remember: true };
+  } else {
+    try {
+      const { data } = await utils.openPrompt<{
+        confirmed: boolean;
+        remember: boolean;
+      }>({
+        origin: message.origin,
+        type: "lnurlAuth",
+        args: {
+          ...message.args,
+          lnurlDetails,
+        },
+      });
+      loginStatus = data;
+    } catch (e) {
+      // user rejected
+      return { error: e instanceof Error ? e.message : e };
+    }
+  }
+
+  // if the user confirmed (or if we already had a publisher with lnurl auth enabled) we perform the authentication
+  if (loginStatus.confirmed) {
+    let authResponse;
+    try {
+      // Sign the message and do the authentication request to the service
+      authResponse = await auth(lnurlDetails);
+    } catch (e) {
+      console.log("LNURL-auth failed");
+      console.error(e);
+      if (e instanceof Error) {
+        PubSub.publish(`lnurl.auth.failed`, {
+          error: e.message,
+          lnurlDetails,
+          origin: message.origin,
+        });
+
+        return { error: e.message };
+      }
+    }
+
+    // if the service returned with a HTTP 200 we still check if the response data is OK
+    if (authResponse?.data.status.toUpperCase() !== "OK") {
+      PubSub.publish(`lnurl.auth.failed`, {
+        authResponse: authResponse,
+        lnurlDetails,
+        origin: message.origin,
+      });
+      return { error: authResponse?.data?.reason };
+    }
+
+    PubSub.publish(`lnurl.auth.success`, {
+      authResponse,
+      lnurlDetails,
+      origin: message.origin,
+    });
+
+    // if auto login should be enabled get the publisher and update the publisher entry
+    if (loginStatus.remember) {
+      allowance = await db.allowances
+        .where("host")
+        .equalsIgnoreCase(message.origin.host)
+        .first();
+      if (allowance?.id) {
+        await db.allowances.update(allowance.id, {
+          lnurlAuth: true,
+        });
+      }
+      await db.saveToStorage();
+    }
+    return { data: authResponse.data };
+  }
+}
+
+/*
+  Execute the LNURL auth
+  returns the response of the LNURL-auth login request
+   or throws an error
+*/
+async function auth(lnurlDetails: LNURLDetails) {
+  if (lnurlDetails.tag !== "login")
+    throw new Error(
+      `LNURL-AUTH FAIL: incorrect tag: ${lnurlDetails.tag} was used`
+    );
+
+  const connector = await state.getState().getConnector();
+
+  // TOOD: the signMessage call can fail
```

Comment:
Do we have an existing issue for this? If yes, can it be linked? If not do we want to create one and link it?

Reply:
no, should this be written down? 
it should not fail, but technically it can... :)

Reply:
me neither. maybe I should just make this a NOTE. ?